Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -278,11 +278,21 @@ infix operator ++: AdditionPrecedence
 public extension Tensor {
     /// Returns a transposed tensor, with dimensions permuted in the specified order.
     @inlinable
+    @differentiable(
+        wrt: self,
+        vjp: _vjpTransposed(withPermutations:) where Scalar: TensorFlowFloatingPoint)
+    func transposed(permutation: Tensor<Int32>) -> Tensor {
+        Raw.transpose(self, perm: permutation)
+    }
+
+    /// Returns a transposed tensor, with dimensions permuted in the specified order.
+    @available(*, deprecated, renamed: "transposed(permutation:)")
+    @inlinable
     @differentiable(
         wrt: self,
         vjp: _vjpTransposed(withPermutations:) where Scalar: TensorFlowFloatingPoint)
```

Comment:
Remove the `vjp: _vjpTransposed(withPermutations:)`  from the `@differentiable` attribute on the old method, since it can be differentiated through. Then the `@differentiable` attribute can be on a single line:

```swift
@differentiable(wrt: self where Scalar: TensorFlowFloatingPoint)
```

Reply:
Very good.